util/db/user/user.go

```diff
@@ -32,3 +32,37 @@ func UserById(ctx context.Context, id string, db *pgxpool.Pool) (map[string]stri
 		"fullName": fullName,
 	}, nil
 }
+
+// CheckEmail checks if an email exists in users table
+func CheckEmail(ctx context.Context, email string, db *pgxpool.Pool) (bool, error) {
```

Comment:
duplicate function.
Use `IsNewUser`. It's an OR check. Email existence can be verified using that itself.